jvm-packages/xgboost4j/src/main/java/ml/dmlc/xgboost4j/java/IRabitTracker.java

```diff
@@ -0,0 +1,20 @@
+package ml.dmlc.xgboost4j.java;
+
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Interface for Rabit tracker implementations with three public methods:
+ *
+ *  - start(): Start the rabit tracker awaiting for worker connections.
+ *  - getWorkerEnvs(): Return the environment variables needed to initialize Rabit clients.
+ *  - waitFor(): Wait for the task execution by the worker nodes.
+ */
+public interface IRabitTracker {
+  Map<String, String> getWorkerEnvs();
+  boolean start();
+  // for Scala-implementation compatibility
+  boolean start(long timeout, TimeUnit unit);
+  int waitFor();
+  int waitFor(long timeout, TimeUnit unit);
```

Comment:
There are in fact three timeouts.

`start(timeout, unit)` times out for binding to a valid IP address/port (although this rarely happens.)
`waitFor(timeout, unit)` times out waiting for the training tasks.
And the timeout value in the Scala's RabitTracker's constructor times out waiting for workers to connect (typically indicative of network issues, e.g. firewall).

Maybe I should drop the first timeout value, and move the constructor timeout to the `start()` function.